File: iniciante/python/1164.py
# -*- coding: utf-8 -*-

N = int(input())
while N < 1 or N > 20:
    N = int(input())
    
for i in range(0, N, 1):
    X = int(input())
    while X < 1 or X > 100000000:
        X = int(input())
    soma_dos_divisores = 0
    for j in range(1, X, 1):
        if X % j == 0:
            soma_dos_divisores += j;
    if soma_dos_divisores == X:
	    print("%d eh perfeito" %X)
    else:
	    print("%d nao eh perfeito" %X)

File: iniciante/c/1164.c
#include <stdio.h>

int main(){

	int N, soma_dos_divisores, i,j, X;
	scanf("%d", &N);
	for (i=0; i < N; i++){
		scanf("%d", &X);
		soma_dos_divisores = 0;
		for (j=1; j <X; j++){
			if (X % j == 0){
				soma_dos_divisores += j;
			}
		}
		if (soma_dos_divisores == X){
			printf("%d eh perfeito\n",X);
		}else{
			printf("%d nao eh perfeito\n",X);
		}
	}
}